### DISCUSSION TITLE
Create Excel Output with Autofilter

### DISCUSSION THREAD

**Discussion by Burn1n9m4n** [kedro-org/kedro#1380] [Idea]
Hello all! I'm trying to output an Excel DataSet (.xlsx) to a gateway folder as a report. I want to provide the entire dataset within the Excel file, but I want the file to have the column filters already on with the settings already handled before the user opens the file. I know that xlsxwriter has the autofilter function along with a host of options to handle those, but is there a smooth way to add those into a kedro pipeline via either a node or within the catalog itself? 

To be more specific, the output file will have the dates for some event. I want the date of the event to be filtered to anything that is happening today. However, if the user decided to broaden the scope, they would have all the data present already.

**Comment by datajoely** [comment]
Hi @Burn1n9m4n as @avan-sh suggests the two ways to do this are [Hooks](https://kedro.readthedocs.io/en/latest/07_extend_kedro/02_hooks.html) and [Custom DataSets](https://kedro.readthedocs.io/en/stable/07_extend_kedro/03_custom_datasets.html)

I'm leaning towards sub-classing the default `pandas.ExcelDataSet` and overriding the [_save()](https://kedro.readthedocs.io/en/stable/_modules/kedro/extras/datasets/pandas/excel_dataset.html#ExcelDataSet) method. The only complication is that I don't think the `pd.ExcelWriter` gives you the worksheet object needed to add the [filter column](https://openpyxl.readthedocs.io/en/2.5.14/filters.html), hopefully there is a way of doing this without re-opening the saved output, but that will work too.
